### PR TITLE
Fixes #9197 - Highlight non-default settings and add reset option

### PR DESCRIPTION
**Pull request by GregSutcliffe**
Ok, this is about 80% done, but I need a little JS/CSS help ;)

Highlighting of non-default rows is done, but there's probably a nicer way than using `font-weight:bold`. The APIv2 call to reset a default works great, but I'm not sure how to add a nice small button (like the existing edit button) to the non-default settings which calls the new action.

Also, bonus points if someone helps me figure out how to reset the bold highlighting appropriately when you save an edit ;)

### CODE REVIEW

**Comment by tstrachota** [comment]
I'd add method `is_default?` to the `Setting` model and return it's result with the response of settings controller's update action.

Then, it needs a bit of js tuning. Check out the success callback in `app/assets/javascripts/settings.js:18`. I guess you could add something like this at the end of the function:

``` js
if (parsed.setting.default.is_default)
  $(this).closest('tr').removeClass('modified');
else
  $(this).closest('tr').addClass('modified');
```

and then in the scss:

``` css
.btn-reset {
  display: none;
}

tr.modified {
  font-weight: bold;
  .btn-reset {
    display: inline;
  }
}
```

Reset default button would be:

``` html
<button type="button" class="btn btn-default btn-reset">
```

...or something similar.

File: app/controllers/api/v2/settings_controller.rb [hunk 1]
```diff
@@ -27,6 +27,13 @@ def update
         process_response (@setting.parse_string_value(params[:setting][:value]) && @setting.save)
       end
 
+      api :DELETE, "/settings/:id/", N_("Reset a setting to default")
+      param :id, String, :required => true
+
+      def destroy
+        process_response (@setting.update_attribute(:value, nil))
```
Comment (ohadlevy) [inline app/controllers/api/v2/settings_controller.rb]: please move it to  a method on the setting model, the implementation on how to reset should be hidden in the model.

File: app/helpers/settings_helper.rb [hunk 2]
```diff
@@ -30,4 +30,8 @@ def short_cat(category)
     category.gsub(/Setting::/,'')
   end
 
+  def is_default?(setting)
+    setting.default == setting.value
```
Comment (ohadlevy) [inline app/helpers/settings_helper.rb]: can probably also be a model method?

File: app/views/settings/index.html.erb [hunk 3]
```diff
@@ -19,7 +19,7 @@
         </thead>
         <tbody>
           <% setting.each do |item| %>
-            <tr>
+            <tr style="<%= is_default?(item) ? 'font-weight:normal' : 'font-weight:bold' %>">
```
Comment (ohadlevy) [inline app/views/settings/index.html.erb]: can you add css class instead of inline style?

**Comment by ohadlevy** [comment]
missing tests?

**Comment by GregSutcliffe** [comment]
@tstrachota updated to your comments - I can't quite figure out how to link the css reset button to the DELETE action though... any advice?

@ohadlevy sure, will add some once I stop fighting with JS :)

**Comment by dLobatog** [comment]
[test] because of the latest enabled cops

**Comment by tbrisker** [comment]
@GregSutcliffe,
Thank you very much for your efforts! 
This PR has been open for a very long time with no activity. 
Please indicate if you plan to continue working on it or not; if not, feel free to close it. 
If you do not respond either way within a couple of weeks, I will assume this PR is abandoned and will go ahead and close it for you.

**Comment by GregSutcliffe** [comment]
Thanks for the poke @tbrisker - the issue is tracked and I'm super busy with community stuff, so lets drop it for now. I'm sure we'll get to it sooner or later ;)

**Pull request closed without merging**